src/main/java/randoop/generation/Bloodhound.java

```diff
@@ -0,0 +1,210 @@
+ *   <li>Interval for recomputing branch coverage information.
+ *   <li>Default weight for cases where a method has zero branches or computed weight is zero.
+ * </ul>
+ */
+public class Bloodhound implements TypedOperationSelector {
+
+  /**
+   * Map of methods under test to their weights. These weights are dynamic and depend on branch
+   * coverage.
+   */
+  private final Map<TypedOperation, Double> methodWeights = new HashMap<>();
+
+  /**
+   * Map of methods under test to the number of times they have been recently selected by the {@link
+   * ForwardGenerator} to construct a new sequence. This map is cleared every time branch coverage
+   * is recomputed.
+   */
+  private final Map<TypedOperation, Integer> methodSelections = new HashMap<>();
+
+  /**
+   * Map of methods under test to the number of times they have ever been selected by the {@link
+   * ForwardGenerator} to extend an existing sequence to construct a new sequence. This definition
+   * is the same as that of {@code methodSelections} except that we do not clear this map every time
+   * we recompute branch coverage. Thus, the integer value for a given method is non-decreasing
+   * during a run of Randoop. The GRT paper does not state its definition of the "number of
+   * invocations" of a method under test.
+   */
+  private final Map<TypedOperation, Integer> methodSuccessfulCalls = new HashMap<>();
+
+  /** List of operations, identical to ForwardGenerator's operation list. */
+  private final SimpleArrayList<TypedOperation> operationSimpleList;
+
+  /**
+   * Hyper-parameter for balancing branch coverage and number of times a method was chosen. The name
+   * alpha is from the GRT paper.
+   */
+  private final double alpha = 0.7;
+
+  /**
+   * Hyper-parameter for decreasing weights of methods between updates to coverage information. The
+   * name p is from the GRT paper.
+   */
+  private final double p = 0.5;
+
+  /** How often to recompute branch coverage, in steps. */
+  private final int branchCoverageInterval = 100;
+
+  /** Maximum number of successful calls so far to any method under test. */
+  private int maxSuccessfulCalls = 0;
+
+  /**
+   * The number of times {@code step()} has been invoked in {@link ForwardGenerator} to construct a
+   * new sequence.
+   */
+  private int stepNum = 0;
+
+  /**
+   * Construct a new instance of Bloodhound, which will choose from among the given operations.
+   *
+   * @param operations list of operations to copy
+   */
+  public Bloodhound(List<TypedOperation> operations) {
+    this.operationSimpleList = new SimpleArrayList<>(operations);
+  }
+
+  /**
+   * The branch coverage information for all methods under test is updated at every {@code
+   * branchCoverageInterval}'th call of this method.
+   */
+  private void updateBranchCoverageMaybe() {
+    stepNum += 1;
+    if (stepNum % branchCoverageInterval == 0) {
+      methodSelections.clear();
+      CoverageTracker.instance.collect();
```

Comment:
What is the relationship between CoverageTracker.instance.collect() and updateWeightsForOperations() which is defined immediately below?  That was not clear to me.